### PR TITLE
Fix Android 10 issue for Scoped Storage

### PR DESCRIPTION
**Pull request by polokoio456**

### CODE REVIEW

File: BigImageViewer/src/main/java/com/github/piasy/biv/view/BigImageView.java [hunk 1]
```diff
@@ -317,19 +325,61 @@ public void saveImageIntoGallery() {
...
+                    boolean saved = bitmap.compress(Bitmap.CompressFormat.PNG, 100, outputStream);
+                    if (mImageSaveCallback != null) {
+                        if (saved) {
+                            mImageSaveCallback.onSuccess(imageUri.toString());
+                        } else {
+                            mImageSaveCallback.onFail(new RuntimeException("saveImageIntoGallery fail"));
+                        }
+                    }
+
+                    if (outputStream != null) {
```
Comment (Piasy) [inline BigImageViewer/src/main/java/com/github/piasy/biv/view/BigImageView.java:351]: Should we flush and close it before firing callback?

**Comment by Piasy** [comment]
Hi @polokoio456 , thanks for your contribution!

Besides my comment above, I'd also know if the new `saveImageIntoGallery` method could be called on new thread? both on/above/below Android Q. Because you add disk IO and compression here.

File: BigImageViewer/src/main/java/com/github/piasy/biv/view/BigImageView.java (hunk 1 above)
Comment (polokoio456) [reply BigImageViewer/src/main/java/com/github/piasy/biv/view/BigImageView.java]: Yes, you are right, I correct it.

**Comment by polokoio456** [comment]
> Hi @polokoio456 , thanks for your contribution!
> 
> Besides my comment above, I'd also like to know if the new `saveImageIntoGallery` method could be called on new thread? both on/above/below Android Q. Because you add disk IO and compression here.

Yes, `saveImageIntoGallery` is better that be called in the work thread.

**Comment by polokoio456** [comment]
> Hi @polokoio456 , thanks for your contribution!
> 
> Besides my comment above, I'd also like to know if the new `saveImageIntoGallery` method could be called on new thread? both on/above/below Android Q. Because you add disk IO and compression here.

Yes, `saveImageIntoGallery` is better that be called in work thread.
I have no idea to let this compression in work thread in a good program in general Java used.
Because I am used to using RxJava to switch the work thread and main thread.
And, I don't want to include RxJava Lib in your lib.

**Comment by Piasy** [comment]
v1.6.3 is released, check it out!

**Pull request merged**